### PR TITLE
Feature/us5378 submit payment

### PR DESCRIPTION
**Pull request by vchtelmakh**
https://rally1.rallydev.com/#/41662702253d/detail/userstory/68862851516

### CODE REVIEW

File: src/app/billing/billing.component.html [hunk 1]
```diff
@@ -15,7 +15,7 @@
   <form [formGroup]="achForm" (ngSubmit)="achNext()">
     <div class="radio-group">
       <label>
-        <input type="radio" formControlName="accountType" [(ngModel)]="gift.accountType" value="personal" checked> Personal
+        <input type="radio" formControlName="accountType" [(ngModel)]="gift.accountType" value="individual" checked> Personal
       </label>&nbsp;&nbsp;
```
Comment (richardbarker) [inline src/app/billing/billing.component.html]: This should stay as "personal"

File: src/app/billing/billing.component.ts [hunk 2]
```diff
@@ -26,7 +33,9 @@ export class BillingComponent implements OnInit {
     private stateManagerService: StateManagerService,
     private gift: GiftService,
     private fb: FormBuilder,
-    private paymentService: ExistingPaymentInfoService) { }
+    private paymentService: ExistingPaymentInfoService,
+    private pmtService: PaymentService,
```
Comment (mjvcallibrity) [inline src/app/billing/billing.component.ts]: paymentService and pmtService? There has to be a better way.

File: src/app/billing/billing.component.ts [hunk 3]
```diff
@@ -103,4 +148,5 @@ export class BillingComponent implements OnInit {
     this.router.navigateByUrl(this.stateManagerService.getNextPageToShow(this.stateManagerService.billingIndex));
   }
 
+
```
Comment (mjvcallibrity) [inline src/app/billing/billing.component.ts]: Why the extra whitespace?

File: src/app/donation/donation.component.ts [hunk 4]
```diff
@@ -10,6 +10,8 @@ export class DonationComponent implements OnInit {
 
   constructor(public gift: GiftService) {}
 
-  ngOnInit() {}
+  ngOnInit() {
+    (<any>window).Stripe.setPublishableKey(process.env.STRIPE_PUBKEY);
```
Comment (mjvcallibrity) [inline src/app/donation/donation.component.ts]: Is there a way to make this a global in the module that is available to any components that need it?

File: src/app/payment/payment.component.ts [hunk 5]
```diff
@@ -23,6 +23,9 @@ export class PaymentComponent implements OnInit {
   }
 
   ngOnInit() {
+
+    (<any>window).Stripe.setPublishableKey(process.env.STRIPE_PUBKEY);
```
Comment (mjvcallibrity) [inline src/app/payment/payment.component.ts]: Is there a way to make this a global in the module that is available to any components that need it?

File: src/app/services/donation.service.ts [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+import { CustomerBank } from '../classes/customer-bank';
+import { CustomerCard} from '../classes/customer-card';
+
+import 'rxjs/add/operator/catch';
+import 'rxjs/add/operator/map';
+
+@Injectable()
+export class DonationService implements Resolve<number> {
+
+  private transactionUrl = process.env.CRDS_API_ENDPOINT;
```
Comment (mjvcallibrity) [inline src/app/services/donation.service.ts]: Does this need to have a type associated with it?

File: src/app/services/donation.service.ts [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+    });
+
+    return observable;
+  }
+
+  updateDonorAndDonate(donorId) {
+
+    if (this.giftService.paymentType === 'cc') {
+
+      this.card =  new CustomerCard('mpcrds+20@gmail.com', 4242424242424242, 12, 17, 123, 12345); //test data
```
Comment (mjvcallibrity) [inline src/app/services/donation.service.ts]: How long is this test data going to live here? When do we remove it?

File: src/app/services/donation.service.ts [hunk 8]
```diff
@@ -0,0 +1,111 @@
...
+      this.paymentService.updateDonorWithCard(donorId, this.card, this.giftService.email)
+        .then(function (donor) {
+          this.donate();
+        }, this.paymentService.stripeErrorHandler);
+
+    } else if (this.giftService.paymentType === 'bank') {
+
+      this.card =  new CustomerCard('mpcrds+20@gmail.com', 4242424242424242, 12, 17, 123, 12345); //test data+
```
Comment (mjvcallibrity) [inline src/app/services/donation.service.ts]: Same here.

File: src/app/services/payment.service.ts [hunk 9]
```diff
@@ -0,0 +1,144 @@
...
+
+        });
+
+        return observable;
+    }
+
+    makeApiDonorCall(donorInfo, email, firstName, lastName, restMethod): Observable<any> {
+
+        let crdsDonor = new CrdsDonorWithoutId(donorInfo.id, email, firstName, lastName);
+
```
Comment (richardbarker) [inline src/app/services/payment.service.ts]: These input parameters need types.  Also there seems to be redundancy here.  donorInfo contains everything already.

File: src/app/billing/billing.component.html (hunk 1 above)
Comment (vchtelmakh) [inline src/app/billing/billing.component.html]: https://stripe.com/docs/stripe.js?

account_holder_type: the type of entity that holds the account. Can be "individual" or "company"

**Comment by richardbarker** [comment]
I like app/models instead of app/classes.  Seems more descriptive and typical of most projects.

File: src/app/services/donation.service.ts (hunk 8 above)
Comment (sseissiger) [inline src/app/services/donation.service.ts]: donation.service not being used yet, we will clean up this sprint when we wire up donations path

File: src/app/services/donation.service.ts (hunk 7 above)
Comment (sseissiger) [inline src/app/services/donation.service.ts]: donation.service not being used yet, we will clean up this sprint when we wire up donations path

File: src/app/payment/payment.component.ts (hunk 5 above)
Comment (sseissiger) [inline src/app/payment/payment.component.ts]: Now setting key in one place - the stripe.service

File: src/app/billing/billing.component.html (hunk 1 above)
Comment (richardbarker) [inline src/app/billing/billing.component.html]: Ok we should probably change the current accountType of "business" to "company" as well, then.

File: src/app/services/payment.service.ts [hunk 10]
```diff
@@ -0,0 +1,140 @@
...
+            put: 'PUT'
+        };
+    }
+
+    //Get Crds Donor by email
+    getDonor(email: string): Observable<any> {
+
+        let encodedEmail = email ? encodeURI(email).replace(/\+/g, '%2B') : '';
+        let donorUrl = this.baseUrl + 'donor/email=' + encodedEmail;   
+        let requestOptions: any = this.httpClient.getRequestOption();
```
Comment (richardbarker) [inline src/app/services/payment.service.ts]: This is not necessary.  Just FYI, options is not required and httpClient will automatically put authorization headers for you.

**Pull request merged**